bat/ASML.BAT

```diff
@@ -1,26 +1,31 @@
-rem asml assemble and link to generate load module 390 from mlc
-echo off
-if %1 == tron ECHO ON
-if %1 == tron shift
-if %1 == TRON ECHO ON
-if %1 == TRON shift
-if exist %1.BAL erase %1.BAL
-if exist %1.PRN erase %1.PRN
-if exist %1.OBJ erase %1.OBJ
-if exist %1.LST erase %1.LST
-if exist %1.390 erase %1.390
-if exist %1.ERR erase %1.ERR
-if exist %1.STA erase %1.STA
-if exist %1.TR* erase %1.TR*
-call %~dps0mz390 %1  sysmac(+mac) syscpy(+mac) %2 %3 %4 %5 %6 %7 %8 %9
-if errorlevel 5 goto mzerr
-call %~dps0lz390 %1 %2 %3 %4 %5 %6 %7 %8 %9
-if errorlevel 5 goto lzerr
-goto end
-:mzerr
-pause see errors on mz390 generated bal file and console
-goto end
-:lzerr
-pause see errors on lz390 generated lst file and console
-erase %1.390
-:end
+rem asml assemble and link to generate load module 390 from mlc
+echo off
+if %1 == tron ECHO ON
+if %1 == tron shift
+if %1 == TRON ECHO ON
+if %1 == TRON shift
+if exist %1.BAL erase %1.BAL
+if exist %1.PRN erase %1.PRN
+if exist %1.OBJ erase %1.OBJ
+if exist %1.LST erase %1.LST
+if exist %1.390 erase %1.390
+if exist %1.ERR erase %1.ERR
+if exist %1.STA erase %1.STA
+if exist %1.TR* erase %1.TR*
+
+rem get the z390 directory
+set "zdir=%~dps0..\"
+for %%f in ("%zdir%") do set "zdir=%%~ff"

```

Comment:
Just so I get this, this command is setting the zdir folder to the parent folder?? So this assumes a specific folder structure when running your assembles..

That is:

    project
    -- code (where the mlc lives)
    -- mac (for any referenced macros or includes)

Just want to understand as I think this should be documented somewhere.. I'm looking at the doco. I assume the

Reply:
This may be more than you wanted to know.

Yes, the z390 software assumes a particular directory hierarchy (the hierarchy used in github). The z390 bat files are assumed to be in the bat subdirectory of the z390 root directory, z390 macros in the mac subdirectory, etc. The variable %zdir% is first set to a relative path for one directory up; the "for" statement converts it to a full path name. A user's mlc files would NOT reside within the z390 root directory. For example, if user fred has file C:\fred\source\MYPGM.MLC, fred could assemble and link it using
    cd <full path to the z390 root directory>
    bat\asml C:\fred\source\mypgm

Line 20 of ASML.BAT

call %~dps0mz390 %1  sysmac(+%zdir%mac) syscpy(+%zdir%mac) %2 %3 %4 %5 %6 %7 %8 %9

will correctly set the sysmac and syscpy values to full path names to the z390 mac directory prefixed with "+", which is a signal to concatenate. (If I remember correctly, the full path directory name for %1 will be the first part of the final sysmac value used, followed by the above value. Abe can keep me honest here.)

You will also notice that line 20 prefixes mz390 with %~dps0, so z390 assumes that MZ390.BAT is in the bat directory.

Looking in MZ390.BAT, line 8, (shortened here)

java -classpath %~dps0..\z390.jar -Xrs -Xms150000K  ...

note the classpath is set to the z390.jar file, which is assumed to be in the z390 root directory. Another use of moving one directory up.

These assumptions (z390.jar in z390 root directory, names and locations of subdirectories) are used throughout the bat files and the newer bash files.

While a developer might be concerned about these assumptions, I'm not sure a user would be.

Hope this answers your questions.

On another note, I'd like to learn how you included the commented ASML.BAT lines.